lib/plugins/aws/package/compile/events/apiGateway/lib/hack/updateStage.js

```diff
@@ -198,32 +202,43 @@ function handleLogs() {
 
 function handleTags() {
   const provider = this.state.service.provider;
-  const tagsMerged = Object.assign({}, provider.stackTags, provider.tags);
-
-  const tagsToCreate = _.entriesIn(tagsMerged).map(pair => ({
-    key: String(pair[0]),
-    value: String(pair[1]),
-  }));
-  if (tagsToCreate) {
-    tagsToCreate.forEach((tag) => {
-      const operation = { op: 'replace', path: `/variables/${tag.key}`, value: tag.value };
-      this.apiGatewayStagePatchOperations.push(operation);
+  const tagsMerged = _.mapValues(Object.assign({}, provider.stackTags, provider.tags),
+      v => String(v));
+  const currentTags = this.apiGatewayStageState.tags || {};
+  const tagKeysToBeRemoved = Object.keys(currentTags)
+    .filter(currentKey => !_.isString(tagsMerged[currentKey]));
+
+  const restApiId = this.apiGatewayRestApiId;
+  const stageName = this.options.stage;
+  const region = this.options.region;
+  const resourceArn = `arn:aws:apigateway:${region}::/restapis/${restApiId}/stages/${stageName}`;
```

Comment:
Does this work with a custom API name? (see `apiName` in https://github.com/serverless/serverless/blob/master/docs/providers/aws/guide/serverless.yml.md).

Reply:
Makes sense :+1:

Just want to be 100% sure that we don't re-introduce any regression here.